test/functionalTest/harnessFunctions.sh

```diff
@@ -771,6 +862,18 @@ function accumulatorStart()
 
 
 
+# ------------------------------------------------------------------------------
+#
+# mqttAccumulatorDump
+#
+function mqttAccumulatorDump()
+{
+  # The dump message is not necessary but nice to have to emphasize intention
+  mosquitto_pub -h $bindIp -t dump_mqtt -m "dump"
+}
```

Comment:
I'm realizing this is not going to work.

accumulatorDump() get the output in the call to the accumulator API, which is based in HTTP so it's synchronous (in other words, the response contains the dump). However, in the case of MQTT, just publishing on a given topic (dump_mqtt topic) is not enough: the testHarness.sh script should keep listening at MQTT in asynchronous way to get the response published under some other topic and this is hard to do.

A best approach is probably include the MQTT listening functionality now in accumulator-mqtt.py in accumulator-server.py, preserving the HTTP based API to get the dump. I mean, accumulator-server.py would be able to listen to both HTTP notifications and MQTT notification (some flags in the CLI will be introduced for that), but the way of getting the notification dump will be the same (HTTP based API, wrapped by accumulatorDump() in harnessFunctions.sh)

Reply:
Fixed by PR https://github.com/telefonicaid/fiware-orion/pull/3889